### PR TITLE
tee: optee: log message if dynamic shm is enabled

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
@jforissier fyi

File: drivers/tee/optee/core.c [hunk 1]
```diff
@@ -223,8 +223,10 @@ static void optee_get_version(struct tee_device *teedev,
 	};
 	struct optee *optee = tee_get_drvdata(teedev);
 
-	if (optee->sec_caps & OPTEE_SMC_SEC_CAP_DYNAMIC_SHM)
+	if (optee->sec_caps & OPTEE_SMC_SEC_CAP_DYNAMIC_SHM) {
 		v.gen_caps |= TEE_GEN_CAP_REG_MEM;
+		pr_info("Dynamic shared memory is enabled and used\n");
```
Comment (jforissier) [inline drivers/tee/optee/core.c]: "enabled" sounds good, but "used" depends on `tee-supplicant`...

Comment (vchong) [reply drivers/tee/optee/core.c]: `and used` removed.

**Comment by jenswikl** [comment]
Could we print this when the driver is initialized instead?

**Comment by vchong** [comment]
Something like this?

```
diff --git a/drivers/tee/optee/core.c b/drivers/tee/optee/core.c
index 7b8f163c40b4..cfdf3ef869db 100644
--- a/drivers/tee/optee/core.c
+++ b/drivers/tee/optee/core.c
@@ -635,6 +635,10 @@ static struct optee *optee_probe(struct device_node *np)
        optee_enable_shm_cache(optee);
 
        pr_info("initialized driver\n");
+
+       if (optee->sec_caps & OPTEE_SMC_SEC_CAP_DYNAMIC_SHM)
+               pr_info("Dynamic shared memory is enabled\n");
+
        return optee;
 err:
        if (optee) {
```

**Comment by jenswikl** [comment]
Yes please.

**Comment by vchong** [comment]
Done. Also added `Suggested-by` for Jerome.

**Comment by jforissier** [comment]
Capitalization: `s/D/d/` for consistency with other messages.

Also, it would make more sense to move the message before "initialized driver" I think.

**Comment by jenswikl** [comment]
Thanks. Please replace `tee: shm` with `tee: optee`

**Comment by vchong** [comment]
Done and done.

**Comment by vchong** [comment]
And done.

**Comment by jenswikl** [comment]
Anyone want's to add a tag before I merge this into my `optee-next` branch? @jforissier, @etienne-lms @jbech-linaro?

**Comment by jforissier** [comment]
Well since I suggested adding the message I should probably ack it:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by vchong** [comment]
Tag added. Thanks!

**Pull request merged**